subprojects/gradle/instrumentation-tests/src/integTest/kotlin/com/avito/instrumentation/reservation/client/kubernetes/KubernetesReservationClientIntegrationTest.kt

```diff
@@ -0,0 +1,173 @@
+
+import com.avito.instrumentation.reservation.adb.AndroidDebugBridge
+import com.avito.instrumentation.reservation.adb.EmulatorsLogsReporter
+import com.avito.instrumentation.reservation.client.ReservationClient
+import com.avito.instrumentation.reservation.request.Device.CloudEmulator
+import com.avito.instrumentation.reservation.request.Reservation
+import com.avito.instrumentation.reservation.request.createStubInstance
+import com.avito.runner.service.worker.device.adb.Adb
+import com.avito.truth.assertThat
+import com.avito.utils.gradle.KubernetesCredentials
+import com.avito.utils.gradle.createKubernetesClient
+import com.avito.utils.logging.FakeCILogger
+import com.google.common.truth.Truth.assertThat
+import io.fabric8.kubernetes.client.KubernetesClientException
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import java.io.File
+import java.net.UnknownHostException
+
+internal class KubernetesReservationClientIntegrationTest {
+
+    private val logger = FakeCILogger()
+    private val deploymentNameGenerator = FakeDeploymentNameGenerator()
+    private var clientOne: ReservationClient? = null
+    private var clientTwo: ReservationClient? = null
+
+    @Test
+    fun `claim - throws exception - unknown host`() {
+        clientOne = createClient(
+            kubernetesUrl = "unknown-host",
+            kubernetesNamespace = "emulators"
+        )
+
+        val exception = assertThrows<KubernetesClientException> {
+            runBlocking {
+                clientOne!!.claim(
+                    reservations = listOf(
+                        Reservation.Data(
+                            device = CloudEmulator.createStubInstance(),
+                            count = 1
+                        )
+                    ),
+                    scope = this
+                )
+            }
+        }
+
+        assertThat<KubernetesClientException>(exception) {
+            assertThat(message).contains(
+                "Operation: [create]  " +
+                    "for kind: [Deployment]  " +
+                    "with name: [null]  " +
+                    "in namespace: [emulators]  " +
+                    "failed"
+            )
+            assertThat<UnknownHostException>(cause) {
+                assertThat(message).contains("unknown-host")
+            }
+        }
+    }
+
+    /**
+     * see MBS-8662
+     */
+    @Test
+    fun `claim - throws exception - deployment already exists`() {
+        clientOne = createClient()
+        clientTwo = createClient()
+
+        val delayEnoughToCreateFirstDeploymentMs = 3000L
+
+        val exception = assertThrows<KubernetesClientException> {
+            runParallel(
```

Comment:
Could we write a test without parallelism and delay?

Reply:
i don't know how to test such thing in integration

Reply:
I agree that we need a real interaction, but I don't know do we need it parallel?
If we use the synchronous dispatcher will that help you?

Reply:
i need two different clients (look at them as different PR jobs) to create deployment with same name one after another;
i'm not sure if sync dispatcher will help me;
if you know how to do it - please help